collector/exporter/otelarrowexporter/internal/arrow/exporter.go

```diff
@@ -228,6 +234,8 @@ func (e *Exporter) SendAndWait(ctx context.Context, data any) (bool, error) {
 		select {
 		case <-ctx.Done():
 			err = ctx.Err()
+		case <-e.shutdown:
+			err = context.Canceled
```

Comment:
The test flake was related to this code here. Shutdown does not cancel the request context, so we needed another signal to cleanly exit the exporter.